fastlane_bot/helpers/txhelpers.py

```diff
@@ -457,47 +464,21 @@ def build_transaction_with_gas(
 
         returns: the transaction to be submitted to the blockchain
         """
+        value = src_amt if (src_address == self.ConfigObj.NATIVE_GAS_TOKEN_ADDRESS and self.ConfigObj.SELF_FUND) else None
```

Comment:
1. Please use `0` instead of `None`; it is always better to maintain the same type for a given variable, unless there really is no value of that type which is suitable for the given scenario; here, passing `value = 0` to the contract function means that it doesn't take any ETH, which is what this conditions is designated for.
2. Something doesn't feel right about the "`and`" here; for example, when the source token is ETH and self-funding is disabled, you pass `value = 0` despite the fact that an ETH amount larger than 0 seems to be needed

Reply:
Updated - please take a look at relevant changes